test/common/common.go

```diff
@@ -0,0 +1,472 @@
+			serviceAddress = lbHostname
+			ips, err := LookupARecord(lbHostname, googleDNSServer)
+			if err != nil {
+				t.Logf("Waiting for IP of loadbalancer %s", lbHostname)
+				// If the hostname cannot be resolved currently then retry later
+				return false, nil
+			}
+			for _, ip := range ips {
+				serviceResolvedIPs[ip] = struct{}{}
+			}
+		} else {
+			t.Logf("Waiting for loadbalancer details for service %s", svcName.Name)
+			return false, nil
+		}
+		t.Logf("Loadbalancer's IP(s): %v", serviceResolvedIPs)
+		return true, nil
+	}); err != nil {
+		return "", nil, fmt.Errorf("failed to get loadbalancer IPs for service %s/%s: %v", svcName.Name, svcName.Namespace, err)
+	}
+
+	return serviceAddress, serviceResolvedIPs, nil
+}
+
+func EnsureOperandResources(ctx context.Context) error {
+	if os.Getenv(e2eSeparateOperandNsEnvVar) != "true" {
+		return nil
+	}
+
+	if err := ensureOperandNamespace(ctx); err != nil {
+		return fmt.Errorf("failed to create %s namespace: %v", operandNamespace, err)
+	}
+
+	if err := ensureOperandRole(ctx); err != nil {
+		return fmt.Errorf("failed to create role external-dns-operator in ns %s: %v", operandNamespace, err)
+	}
+
+	if err := ensureOperandRoleBinding(ctx); err != nil {
+		return fmt.Errorf("failed to create rolebinding external-dns-operator in ns %s: %v", operandNamespace, err)
+	}
+
+	return nil
+}
+
+func ensureOperandNamespace(ctx context.Context) error {
+	return KubeClient.Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: operandNamespace}})
+}
+
+func ensureOperandRole(ctx context.Context) error {
+	rules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"secrets", "serviceaccounts", "configmaps"},
+			Verbs:     []string{"get", "list", "watch", "create", "update", "delete"},
+		},
+		{
+			APIGroups: []string{""},
+			Resources: []string{"namespaces"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"apps"},
+			Resources: []string{"deployments"},
+			Verbs:     []string{"get", "list", "watch", "create", "update", "delete"},
+		},
+	}
+
+	role := rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rbacRsrcName,
+			Namespace: operandNamespace,
+		},
+		Rules: rules,
+	}
+	return KubeClient.Create(ctx, &role)
+}
+
+func ensureOperandRoleBinding(ctx context.Context) error {
+	rb := rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rbacRsrcName,
+			Namespace: operandNamespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     rbacRsrcName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      operatorServiceAccount,
+				Namespace: OperatorNamespace,
+			},
+		},
+	}
+	return KubeClient.Create(ctx, &rb)
+}
+
+func initKubeClient() error {
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		return fmt.Errorf("failed to get kube config: %v", err)
```

Comment:
The approach taken in `initScheme` with a `panic` seems to be logical for an init function. WDYT about aligning them?